bookkeeper-server/src/test/java/org/apache/bookkeeper/suites/BookKeeperClusterTestSuite.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.bookkeeper.suites;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.bookkeeper.client.BookKeeperAdmin;
+import org.apache.bookkeeper.common.net.ServiceURI;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.conf.TestBKConfiguration;
+import org.apache.bookkeeper.proto.BookieServer;
+import org.apache.bookkeeper.test.PortManager;
+import org.apache.bookkeeper.test.TestStatsProvider;
+import org.apache.bookkeeper.util.IOUtils;
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+/**
+ * A class runs a bookkeeper cluster for testing.
+ *
+ * <p>The cluster will be setup and teardown before class. It will not be restarted between different
+ * test methods. It is more suitable for running tests that don't require restarting bookies.
+ */
+@Slf4j
+public abstract class BookKeeperClusterTestSuite {
+
+    protected static MetadataStore metadataStore;
+    protected static ClientConfiguration baseClientConf;
+    protected static ServerConfiguration baseServerConf;
+    protected static final int NUM_BOOKIES = 3;
+    protected static final List<BookieServer> BOOKIES = new ArrayList<>(NUM_BOOKIES);
+    protected static final List<File> TMP_DIRS = new ArrayList<>(NUM_BOOKIES);
+
+    protected static File createTempDir(String prefix, String suffix) throws IOException {
+        File dir = IOUtils.createTempDir(prefix, suffix);
+        TMP_DIRS.add(dir);
+        return dir;
+    }
+
+    protected static ServerConfiguration newServerConfiguration() throws Exception {
+        File f = createTempDir("bookie", "test");
+        int port = PortManager.nextFreePort();
+        return newServerConfiguration(port, f, new File[] { f });
+    }
+
+    protected static ServerConfiguration newServerConfiguration(int port, File journalDir, File[] ledgerDirs) {
```

Comment:
hmm..we can reuse BookKeeperClusterTestCase.newServerConfiguration. Probably make this as static method and move it to some test utility class, so that it can be used in both the places